lib/SILOptimizer/Mandatory/MoveOnlyUtils.cpp

```diff
@@ -253,6 +253,9 @@ bool noncopyable::memInstMustReinitialize(Operand *memOper) {
     auto *CAI = cast<ExplicitCopyAddrInst>(memInst);
     return CAI->getDest() == address && !CAI->isInitializationOfDest();
   }
+  case SILInstructionKind::MarkDependenceAddrInst: {
+    return true;
```

Comment:
IIUC memInstMustReinitialize should only be true for the `address` operand of mark_dependence_addr. I don't think a simple read of the base qualifies.

Reply:
That was my first try. But the move-only checker crashes if it only returns true for `address`.